--- a/src/third_party/devtools-frontend/src/scripts/build/rollup.config.js
+++ b/src/third_party/devtools-frontend/src/scripts/build/rollup.config.js
@@ -18,11 +18,6 @@
     sourcemap: Boolean(commandLineArgs.configSourcemaps),
   }],
   plugins: [
-    terser({
-      compress: {
-        pure_funcs: commandLineArgs.configDCHECK ? ['Platform.DCHECK'] : [],
-      },
-    }),
     {
       name: 'devtools-plugin',
       resolveId(source, importer) {
